UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -322,7 +315,12 @@ public float GetCumulativeReward()
         /// </summary>
         public void Done()
         {
-            m_Done = true;
+            NotifyAgentDone();
+            _AgentReset();
+            m_RequestAction = false;
+            m_RequestDecision = false;
+            m_Reward = 0f;
+            m_CumulativeReward = 0f;
```

Comment:
Feels like this could be moved into NotifyAgentDone() (or maybe combine Done and NotifyAgentDone, unless you don't want to the user to set maxStepReached)

Reply:
Moved some things around